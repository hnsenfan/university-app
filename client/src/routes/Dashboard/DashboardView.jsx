import { Fragment, useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import { END_POINTS } from '../../api'

import UniversityCard from '../Universities/UniversityCard'
import NewsletterModal from '../../components/NewsletterModal/NewsletterModal'

import './DashboardView.scss'

const DashboardView = () => {
  const [isShowModal, setIsShowModal] = useState(false)
  const [favUniversityList, setFavUniversityList] = useState([])

  const closeModal = () => {
    setIsShowModal(false)
  }

  useEffect(async () => {
    const getUserList = async () => {
      const result = await END_POINTS.getUserProfile()
      setIsShowModal(!result.data.subscribe_newsletter)
      setFavUniversityList(result.data.fav_universities)
    }
    getUserList()
  }, [])
  return (
    <div className='page--whole'>
      <section className='container--md'>
        <div className='d-flex justify-content-end radius-m pv-xl student-bg mv-l'>
          <div className='d-flex flex-column p-xl'>
            <h1 className='promotion-title'>Building A Better World One Student At A Time</h1>
            <p className='promotion-desc'>We have matched more than 100,000 students across the world with their dream universities. Find yours on our site now!</p>
            <Link to='/universities'>
              <button className='btn-primary btn-block'>GO</button>
            </Link>
          </div>
        </div>
      </section>
      { favUniversityList && favUniversityList.length > 0 &&
        <Fragment>
          <aside className='container--md mt-s'>
            <label className='text-medium text-medium-grey'>Your Favourite Universities</label>
          </aside>
          {/* Section for Order Details (customer and vendor) */}
          <section className='container--md mt-m'>
            { favUniversityList.map((uni, index) =>
              <UniversityCard key={index} uniObject={uni} isFavouriteProps />
            )}
          </section>
        </Fragment>
      }
      <NewsletterModal show={isShowModal} handleClose={closeModal} />
    </div>
  )
}

export default DashboardView