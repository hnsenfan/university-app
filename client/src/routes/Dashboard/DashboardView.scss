@import '/src/modules/styles/core/variables';

.student-bg {
    background-image: url('../../assets/student.jpeg');
    background-repeat: no-repeat;
    background-position: right;
    background-size: 750px;
    height: 400px
}

.promotion-title {
  text-align: right;
  font-size: 30px;
  max-width: 350px;
  color: map-get($colors, text-green)
}

.promotion-desc {
  max-width: 350px;
}