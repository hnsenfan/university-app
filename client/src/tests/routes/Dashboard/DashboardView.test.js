import { shallow } from 'enzyme'
import DashboardView from '../../../routes/Dashboard/DashboardView'

test('should test DashboardView component', () => {
  const wrapper = shallow(<DashboardView />)
  expect(wrapper).toMatchSnapshot()
 })
