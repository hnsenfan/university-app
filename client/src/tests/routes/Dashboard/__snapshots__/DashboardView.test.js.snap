// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test DashboardView component 1`] = `
<div
  className="page--whole"
>
  <section
    className="container--md"
  >
    <div
      className="d-flex justify-content-end radius-m pv-xl student-bg mv-l"
    >
      <div
        className="d-flex flex-column p-xl"
      >
        <h1
          className="promotion-title"
        >
          Building A Better World One Student At A Time
        </h1>
        <p
          className="promotion-desc"
        >
          We have matched more than 100,000 students across the world with their dream universities. Find yours on our site now!
        </p>
        <Link
          to="/universities"
        >
          <button
            className="btn-primary btn-block"
          >
            GO
          </button>
        </Link>
      </div>
    </div>
  </section>
  <NewsletterModal
    handleClose={[Function]}
    show={false}
  />
</div>
`;
